src/manufacturer/manufacturerDialog.component.tsx

```diff
@@ -207,45 +200,45 @@ function ManufacturerDialog(props: ManufacturerDialogProps) {
   }, [handleErrors, manufacturerDetails, addManufacturer, handleClose]);
 
   const handleEditManufacturer = React.useCallback(() => {
-    if (manufacturerDetails && selectedManufacturerData) {
+    if (manufacturerDetails && selectedManufacturer) {
       const hasErrors = handleErrors();
 
       if (hasErrors) {
         return;
       }
 
       const isNameUpdated =
-        manufacturerDetails.name !== selectedManufacturerData.name;
+        manufacturerDetails.name !== selectedManufacturer?.name;
```

Comment:
```suggestion
        manufacturerDetails.name !== selectedManufacturer.name;
```
Same for the others below, line 203 ensures it isn't undefined.